backend/Dockerfile

```diff
@@ -40,13 +34,41 @@ COPY ${SOURCE_CODE}/ ./
 RUN GO111MODULE=on CGO_ENABLED=1 GOEXPERIMENT=strictfipsruntime go build -tags strictfipsruntime -o /bin/apiserver ./backend/src/apiserver/ && \
     dnf clean all
 
+# 2. Compile preloaded pipeline samples
+FROM registry.access.redhat.com/ubi9/python-39:9.5 as compiler
+USER root
+RUN dnf install -y python3-setuptools jq
+RUN wget https://bootstrap.pypa.io/get-pip.py && python3 get-pip.py
+COPY backend/requirements.txt .
+RUN python3 -m pip install -r requirements.txt --no-cache-dir
+
+# Downloading Argo CLI so that the samples are validated
+ENV ARGO_VERSION v3.5.14
+RUN curl -sLO https://github.com/argoproj/argo-workflows/releases/download/${ARGO_VERSION}/argo-linux-amd64.gz && \
+    gunzip argo-linux-amd64.gz && \
+    chmod +x argo-linux-amd64 && \
+    mv ./argo-linux-amd64 /usr/local/bin/argo
+
+WORKDIR /
+COPY ./samples /samples
+COPY backend/src/apiserver/config/sample_config.json /samples/
+
+# Compiling the preloaded samples.
+# The default image is replaced with the GCR-hosted python image.
+RUN set -e; \
+    < /samples/sample_config.json jq ".pipelines[].file" --raw-output | while read pipeline_yaml; do \
+    pipeline_py="${pipeline_yaml%.yaml}"; \
+    echo "Compiling: \"$pipeline_py\"" && python3 "$pipeline_py" && echo -n "Output: " && ls "$pipeline_py.yaml"; \
+    done
```

Comment:
I don't think we need this, we don't load the sample files in rhoai and odh: 

https://github.com/red-hat-data-services/data-science-pipelines/blob/main/backend/Dockerfile